sphinx_gallery/gen_gallery.py

```diff
@@ -29,6 +29,7 @@
 DEFAULT_GALLERY_CONF = {
     'filename_pattern': re.escape(os.sep) + 'plot',
     'examples_dirs': os.path.join('..', 'examples'),
+    'examples_dirs_order': None,
```

Comment:
I would have a example_dirs_sort_key or something, which would be used like this in conf.py:

```
'example_dirs_sort_key': ExplicitOrderKey(['folder', 'another_folder'])
```

Not by that default, we should still sort alphabetically, but I am guessing this is what `key=None` does.